File: r/nt_toolbox/toolbox_signal/subsampling.R



subsampling <- function(x, d){
  ####
  # subsampling along dimension d by factor p=2
  ####
  p <- 2
  if (d==1){
    y <- x[ ((1:dim(x)[1])%%p == 1 ) , , drop=F]
  }
  else if (d==2){
    y <- x[ , ((1:dim(x)[2])%%p == 1 ) , drop=F]
  }
  else{
    warning('Not implemented')
  }
  return(y)
}

File: matlab/toolbox_signal/subsampling.m
function y = subsampling(x,d,p)

% downsampling - subsampling along dimension d
%
%   y = subsampling(x,d,p);
%
%   default is p==2, d==1
%
%   Copyright (c) 2009 Gabriel Peyre

if nargin<3
    p = 2;
end
if nargin<2
    d = 1;
end

switch d
    case 1
        y = x(1:p:end,:,:);
    case 2
        y = x(:,1:p:end,:);
    case 3
        y = x(:,:,1:p:end);
    otherwise
        error('Not implemented');
end
